src/sqlfluff/core/rules/base.py

```diff
@@ -639,8 +639,7 @@ def indent(self) -> str:
         space = " "
         return space * self.tab_space_size if self.indent_unit == "space" else tab
 
-    def is_final_segment(self, context: RuleContext) -> bool:
-        """Is the current segment the final segment in the parse tree."""
+    def _is_final_segment_helper(self, context: RuleContext):
```

Comment:
I split `is_final_segment()` into two functions in order to extract this reusable helper function, which is also used by the new `closing_ancestors()` function below.